### ISSUE TITLE
Can't select rows when using setState with an array value

### ISSUE THREAD

**Issue by pineconesundae** [react-bootstrap-table/react-bootstrap-table2#789]
This issue is very similar to https://github.com/react-bootstrap-table/react-bootstrap-table2/issues/297, which has been fixed (and works!), but if you try to setState with an array, the issue returns.

Partial example:
```jsx
constructor(props) {
  super(props);
  this.state = { results: [] };
}

select(row) {
  this.setState({
    results: []
  });
}

const selectRow = {
  mode: 'radio',
  clickToSelect: true,
  onSelect: this.select
};
```

If you change `results: []` to `results: ''`, it works fine.

```json
"react-bootstrap-table-next": "2.0.1",
"react-bootstrap-table2-filter": "1.1.2",
"react-bootstrap-table2-paginator": "2.0.1"
```

**EDIT**:
This issue title is a bit lacking, since I just discovered that using an object (`{}`) in the setState also prevents the selection from showing.

**Comment by AllenFang** [comment]
@pineconesundae I will check this out again, thanks,  I just test it again and I found that call setState will break the selection procecss

**Comment by harshzalavadiya** [comment]
Having same issue

Demo: https://stackblitz.com/edit/react-yeuhjg

**Comment by harshzalavadiya** [comment]
Okay so, I tried it with different package versions and feature was working smoothly on initial releases but gone broke on 0.1.4(error) and stopped reflecting checkbox changes after patched in 0.1.5 works perfectly in 0.1.3

**Comment by AllenFang** [comment]
Fixed: https://react-bootstrap-table.github.io/react-bootstrap-table2/blog/2019/02/24/version-bump.html
